src/activity_actor.cpp

```diff
@@ -1124,36 +1124,101 @@ std::unique_ptr<activity_actor> hotwire_car_activity_actor::deserialize( JsonVal
     return actor.clone();
 }
 
-void hacksaw_activity_actor::start( player_activity &act, Character &/*who*/ )
+bool tool_activity_actor::start_finish( map &here, player_activity &act, Character &who,
+                                        int &moves, const std::string &tool_name,
+                                        const cata::value_ptr<activity_data_common> map_data_common_t::* const tool_member, bool finish,
+                                        activity_data_common *&data )
 {
-    const map &here = get_map();
-    int moves_before_quality;
-
     if( here.has_furn( target ) ) {
         const furn_id &furn_type = here.furn( target );
-        if( !furn_type->hacksaw->valid() ) {
+        const auto furn_tool_ad = ( *furn_type.*tool_member );
```

Comment:
I failed in my effort to implement `int_id::operator->*`, so I had to use `*foo.*bar` instead of `foo->*bar`.